packages/xo-server/src/xo-mixins/resource-sets.js

```diff
@@ -174,6 +178,27 @@ export default class {
       set.name = name
     }
     if (subjects) {
+      await Promise.all(
+        difference(set.subjects, subjects).map(async subjectId =>
+          Promise.all(
+            (await this._xo.getAclsForSubject(subjectId)).map(async acl => {
+              try {
+                const object = this._xo.getObject(acl.object)
+                if (object.type === 'VM' && object.resourceSet === id) {
+                  await this._xo.removeAcl(subjectId, acl.object, acl.action)
+                  $defer.onFailure(() =>
+                    this._xo.addAcl(subjectId, acl.object, acl.action)
+                  )
+                }
+              } catch (err) {
+                if (!startsWith(err && err.message, 'no such object')) {
```

Comment:
What if the pool is simply not connected?

Reply:
Then I don't want the operation to fail. If the pool isn't connected, the users will keep their ACLs but I don't know how to properly handle that for now.

Reply:
I agree, let's keep it like that for the moment.